### PR TITLE
Fix docs for the record command

### PR DESCRIPTION
**Pull request by jmargeta**
Renames the non-existent "render" command to "record"

### CODE REVIEW

**Comment by yuvaltassa** [comment]
Thanks!

Obviously this change is good, are there any lessons from the discussion in #127 that we want to incorporate in this PR? cc @kevinzakka

**Comment by jmargeta** [comment]
Great, thanks for the pointer @yuvaltassa
Agree, I updated ffmpeg's arguments to work well with the humanoid example.

There is always a black rectangle on the bottom left which should be the "subsampled depth image"
https://github.com/google-deepmind/mujoco/blob/a00f1b5f3043ed1cf676b130a15a2b2ee7700db3/sample/record.cc#L278
image

I would push one more change to optionally disable the depth map overlay through a command line arg, e.g.:
`record humanoid.xml 5 60 rgb.out 0`

File: doc/programming/samples.rst [hunk 1]
```diff
@@ -156,15 +156,14 @@ command-line arguments. For example, a 5 second animation at 60 frames per secon
 
 .. code-block:: Shell
 
-     render humanoid.xml 5 60 rgb.out
+     record humanoid.xml 5 60 rgb.out
 
 The default humanoid.xml model specifies offscreen rendering with 800x800 resolution. With this information in hand, we
```
Comment (yuvaltassa) [inline doc/programming/samples.rst:161]: 1. 800x800 is [not true](https://github.com/google-deepmind/mujoco/blob/9ac048e262e664b5951678ae5c098d9175862218/model/humanoid/humanoid.xml#L22).
2. The doc should be explicit about the importance `visual/global/{offwidth,offheight}` to be large enough or the command will fail.

File: doc/programming/samples.rst [hunk 2]
```diff
@@ -150,21 +150,53 @@ This code sample simulates the passive dynamics of a given model, renders it off
 values, and saves them into a raw data file that can then be converted into a movie file with tools such as ffmpeg. The
 rendering is simplified compared to :ref:`simulate.cc <saSimulate>` because there is no user interaction, visualization
 options or timing; instead we simply render with the default settings as fast as possible. The dimensions and number of
-multi-samples for the offscreen buffer are specified in the MuJoCo model, while the simulation duration, frames-per-
-second to be rendered (usually much less than the physics simulation rate), and output file name are specified as
-command-line arguments. For example, a 5 second animation at 60 frames per second is created with:
+multi-samples for the offscreen buffer are specified in the MuJoCo model with the visual/global/{offwidth,offheight}
```
Comment (yuvaltassa) [inline doc/programming/samples.rst:153]: No. 

multisamples are specified with 
https://mujoco.readthedocs.io/en/stable/XMLreference.html#visual-quality-offsamples
height and width are specified with
https://mujoco.readthedocs.io/en/stable/XMLreference.html#visual-global-offheight
https://mujoco.readthedocs.io/en/stable/XMLreference.html#visual-global-offwidth

File: doc/programming/samples.rst [hunk 3]
```diff
@@ -150,21 +150,53 @@ This code sample simulates the passive dynamics of a given model, renders it off
...
+
+For example, a 5 second animation at 60 frames per second is created with:
 
 .. code-block:: Shell
 
-     ffmpeg -f rawvideo -pixel_format rgb24 -video_size 800x800
-       -framerate 60 -i rgb.out -vf "vflip" video.mp4
+     record humanoid.xml 5 60 rgb.out
+
+The default humanoid.xml model specifies offscreen rendering with 2560x1440 resolution. With this information in hand,
```
Comment (yuvaltassa) [inline doc/programming/samples.rst:194]: link to the file?
https://github.com/google-deepmind/mujoco/blob/main/model/humanoid/humanoid.xml

File: sample/record.cc [hunk 4]
```diff
@@ -217,8 +217,8 @@ void closeOpenGL(void) {
 
 int main(int argc, const char** argv) {
   // check command-line arguments
-  if (argc!=5) {
-    std::printf(" USAGE:  record modelfile duration fps rgbfile\n");
+  if (argc < 5 || argc > 6) {
+    std::printf(" USAGE:  record modelfile duration fps rgbfile [adddepth]\n");
```
Comment (yuvaltassa) [inline sample/record.cc:221]: prefer `depthoverlay` to `adddepth`

File: doc/programming/samples.rst [hunk 5]
```diff
@@ -150,21 +150,53 @@ This code sample simulates the passive dynamics of a given model, renders it off
...
+we can compress the (large) raw date file into a playable movie file:
```
Comment (yuvaltassa) [inline doc/programming/samples.rst:195]: date -> data

**Review by yuvaltassa — CHANGES_REQUESTED** [review]
Looking good :)

File: doc/programming/samples.rst (hunk 2 above)
Comment (yuvaltassa) [reply doc/programming/samples.rst:153]: (Of course use proper RST links, not raw html)

File: doc/programming/samples.rst (hunk 1 above)
Comment (jmargeta) [reply doc/programming/samples.rst]: 1. Thanks, fixed!
2. I added a little comment. I cannot reproduce "large enough". Works for me even for 4x4 given that the model_size and the offscreen render sizes are identical. Is there anything that I am missing? :)

File: doc/programming/samples.rst (hunk 2 above)
Comment (jmargeta) [reply doc/programming/samples.rst]: Great! Is this what you had in mind? :)
https://github.com/google-deepmind/mujoco/blob/f2873724672e660e6b095384ba54e3a499cd3447/doc/programming/samples.rst#record

File: doc/programming/samples.rst [hunk 6]
```diff
@@ -150,21 +150,58 @@ This code sample simulates the passive dynamics of a given model, renders it off
...
+visual/global/{`offwidth <https://mujoco.readthedocs.io/en/stable/XMLreference.html#visual-global-offwidth>`_,
+`offheight <https://mujoco.readthedocs.io/en/stable/XMLreference.html#visual-global-offheight>`_} and
+visual/quality/`offsamples <https://mujoco.readthedocs.io/en/stable/XMLreference.html#visual-quality-offsamples>`_
+attributes, while the simulation duration, frames-per-second to be rendered (usually much less than the physics
+simulation rate), and output file name are specified as command-line arguments.
 
 .. code-block:: Shell
 
-     render humanoid.xml 5 60 rgb.out
+   record modelfile duration fps rgbfile [depthoverlay]
```
Comment (yuvaltassa) [inline doc/programming/samples.rst:162]: So sorry, `depthoverlay` is unreadable, let's either revert to `adddepth` or just `depth`, up to you :)

File: doc/programming/samples.rst (hunk 2 above)
Comment (yuvaltassa) [reply doc/programming/samples.rst]: Perfect :)

File: doc/programming/samples.rst (hunk 1 above)
Comment (yuvaltassa) [reply doc/programming/samples.rst]: Really? You can record videos that are larger than the height/width spec ??
That's strange. I will look again tomorrow.

Comment (jmargeta) [reply doc/programming/samples.rst]: Not larger nor smaller, but exactly according to the spec - I changed the offwidth and offheight in humanoid.xml to values between 4 and 4000 (same values for ffmpeg) and then could record video with such resolution.

Thank you @yuvaltassa for all your thorough feedback

File: doc/programming/samples.rst (hunk 6 above)
Comment (jmargeta) [reply doc/programming/samples.rst]: Haha, no worries, I reverted to adddepth then.

File: doc/programming/samples.rst [hunk 7]
```diff
@@ -150,21 +150,58 @@ This code sample simulates the passive dynamics of a given model, renders it off
...
+
+.. code-block:: Shell
+
+     record humanoid.xml 5 60 rgb.out
+
+The default `humanoid.xml <https://github.com/google-deepmind/mujoco/blob/main/model/humanoid/humanoid.xml>`_ model specifies offscreen rendering with 2560x1440 resolution. With this information in hand,
+we can compress the (large) raw data file into a playable movie file:
 
 .. code-block:: Shell
+     ffmpeg -f rawvideo -pixel_format rgb24 -video_size 2560x1440
```
[the remark this answers is not in the document]
Comment (jmargeta) [reply doc/programming/samples.rst:201]: Hmmm, indeed and it seems that the window size 800x800 does not matter

I compiled MuJoCo with glfw backend (on both Windows and Linux) and the videos always always render fine at exactly the same resolution as is set in the model file by offheight and offwidth.

Then I recompiled MuJoCo with window 1x1
`GLFWwindow* window = glfwCreateWindow(1, 1, "Invisible window", NULL, NULL);`
...and all is still fine!!

Compiling MuJoCo with GLFW_USE_OS_MESA, or compiling record,cc using GLFW, EGL or OSMESA as backends (by adding #defines and relevant libraries (EGL or OSMesa) to the linker), there still does not seem to be any difference or crash on my machines. The selected code paths are executed and the video just keeps being generated. There is a chance I am missing something 🤷

...all I initially wanted was to rename render to record in docs, haha

**Review by yuvaltassa — APPROVED** [review]
Thanks so much! Sorry for the faff. Obviously we're very grateful that you were okay with working on a PR that was initially a 1-word change 🥇

**Comment by jmargeta** [comment]
With pleasure! I think it is a much better PR than it originally was and I got to learn something new about offscreen rendering as a bonus, so thank you @yuvaltassa =)

**Pull request merged**